mapie/estimator/interface.py

```diff
@@ -0,0 +1,340 @@
+            Input labels.
+
+        train_index: ArrayLike of shape (n_samples_train)
+            Training data indices.
+
+        sample_weight: Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+            By default ``None``.
+
+        Returns
+        -------
+        Tuple[RegressorMixin, NDArray, ArrayLike]
+
+        - [0]: RegressorMixin, fitted estimator
+        - [1]: NDArray of shape (n_samples_val,),
+          estimator predictions on the validation fold.
+        - [2]: ArrayLike of shape (n_samples_val,),
+          validation data indices.
+        """
+
+    @staticmethod
+    def _predict_oof_estimator(
+        estimator: RegressorMixin,
+        X: ArrayLike,
+        val_index: ArrayLike,
+    ):
+        """
+        Perform predictions on a single out-of-fold model on a validation set.
+
+        Parameters
+        ----------
+        estimator: RegressorMixin
+            Estimator to train.
+
+        X: ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        val_index: ArrayLike of shape (n_samples_val)
+            Validation data indices.
+
+        Returns
+        -------
+        Tuple[NDArray, ArrayLike]
+            Predictions of estimator from val_index of X.
+        """
+
+    def _aggregate_with_mask(
+        self,
+        x: NDArray,
+        k: NDArray
+    ) -> NDArray:
+        """
+        Take the array of predictions, made by the refitted estimators,
+        on the testing set, and the 1-or-nan array indicating for each training
+        sample which one to integrate, and aggregate to produce phi-{t}(x_t)
+        for each training sample x_t.
+
+        Parameters
+        ----------
+        x: ArrayLike of shape (n_samples_test, n_estimators)
+            Array of predictions, made by the refitted estimators,
+            for each sample of the testing set.
+
+        k: ArrayLike of shape (n_samples_training, n_estimators)
+            1-or-nan array: indicates whether to integrate the prediction
+            of a given estimator into the aggregation, for each training
+            sample.
+
+        Returns
+        -------
+        ArrayLike of shape (n_samples_test,)
+            Array of aggregated predictions for each testing sample.
+        """
+
+    def _pred_multi(self, X: ArrayLike) -> NDArray:
+        """
+        Return a prediction per train sample for each test sample, by
+        aggregation with matrix ``k_``.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples_test, n_features)
+            Input data
+
+        Returns
+        -------
+        NDArray of shape (n_samples_test, n_samples_train)
+        """
+
+    def predict_calib(self, X: ArrayLike) -> NDArray:
+        """
+        Perform predictions on X : the calibration set. This method is
+        called in the ConformityScore class to compute the conformity scores.
```

Comment:
UPD: this method is not called in the ConformityScore class.

Reply:
Done